stackrox/stackrox
# ROX-25066: Create s3 compatible plugin

### Description



Adds a new S3 compatible plugin.

- Depends on #11877

### User-facing documentation




### Testing


#### Automated testing



  

#### How I validated my change



  - Create a new set of access keys in the [mino-console](https://play.min.io:9443/).
  - Create a new bucket, for example `test-s3`.
  - Use the following go test to test the integration:
```golang
func TestS3Compatible(t *testing.T) {
	integration := &storage.ExternalBackup{
		Name:          "test-s3-compatible",
		Type:          types.S3CompatibleType,
		BackupsToKeep: 2,
		Config: &storage.ExternalBackup_S3Compatible{
			S3Compatible: &storage.S3Compatible{
				Bucket:          "test-s3",
				AccessKeyId:     "<access-key>",
				SecretAccessKey: "<secret-access-key>",
				Region:          "us-east-1",
				ObjectPrefix:    "",
				Endpoint:        "https://play.min.io:9000",
				UrlStyle:        storage.S3URLStyle_S3_URL_STYLE_PATH,
			},
		},
	}

	s3, err := newS3Compatible(integration)
	require.NoError(t, err)
	require.NoError(t, s3.Test())

	buff := bytes2.NewBuffer([]byte("some data here"))
	reader := bufio.NewReader(buff)
	require.NoError(t, s3.Backup(io.NopCloser(reader)))
}
```
  - Check the objects are created (and pruned) in the mino-console.
